### ISSUE TITLE
incompatibility with passport 0.3.0

### ISSUE THREAD

**Issue by marrone** [feathersjs/feathers-passport#6]
This is not technically an issue with the `feathers-passport` code, but rather the `passport` module dependency version it pulls down. And this issue is more to document what I am seeing with hopes someone can better explain.

I include `passport` as a package dependency at the top level of my app to be used elsewhere. `feathers-passport` allows for this dependency injection by passing to the initialization ala:

```
var passport = require("passport");
...
.configure(feathersPassport(function(result) {
    var MongoStore = connectMongo(result.createSession);
    result.passport = passport; //inject my own passport module
...
```
The issue arises when my top-level `passport` module version is the latest 0.3.0, and the version pulled in by `feathers-passport`'s own dependency is 0.2.2. See the change made here:
https://github.com/jaredhanson/passport/commit/09ccc550798ebe94cb41443d50ecb73358546b49

The session initialization is moved from passport initialization to the req module, and feathers-passport still uses it's own version of that module at the older version. The result is a run-time error thrown when trying to authenticate as req._passport.session is undefined.

I guess, my confusion is why feathers-passport ends up with the 0.2.2 version of passport, as the package.json specifies the version using the ^, which as I understand should get the latest 0.* version?

**Comment by marshallswain** [comment]
Try `cd`ing into the node_modules/passport directory and running `npm install passport@0.3.0`.  Does that fix the issue for you?  The tests all pass on passport 0.3.0.

**Comment by marrone** [comment]
Yeah, I am doing that currently, but I guess my question is why isnt 0.3.0 installed already if the dependency version is specified as ^0.2.2?

**Comment by marshallswain** [comment]
I first ran into this with donejs a couple days ago.  I haven't looked into the specifics of what causes it, but for some reason after I upgraded from node 0.12.4 to 0.12.8 when I do npm install I don't always get the latest version of the package.  `npm install passport` currently installs 0.2.2 for me, not 0.3.0.  In order to get 0.3.0, I have to use passport@0.3.0.  I tested downgrading node and the behavior is different.
